CMakeLists.txt

```diff
@@ -52,6 +52,15 @@ execute_process(
 add_definitions(-DSDK_VERSION=\"${GIT_COMMIT_HASH}\")
 add_definitions(-DVERSION_STRING=\"${PROJECT_VERSION}\")
 add_definitions(-DDETECTED_GIT_HASH)
+if(NOT KVS_STACK_SIZE OR KVS_STACK_SIZE STREQUAL "")
```

Comment:
Rather than checking for an empty string, can we validate that it is a numeric value with something like:
```
if(KVS_STACK_SIZE MATCHES "^[+-]?[0-9]+$")
```

Reply:
I'm thinking of removing this section actually. Because when I ran Producer-C with the thread stacksize changes, I was able to pass it in via producer-c's CMake command:
```
$ cmake .. -DBUILD_DEPENDENCIES=OFF -DKVS_DEFAULT_STACK_SIZE=256 -DCONSTRAINED_DEVICE=ON
...
CMake Error at dependency/libkvspic/kvspic-src/CMakeLists.txt:62 (message):
  Conflicting parameters: KVS_DEFAULT_STACK_SIZE and CONSTRAINED_DEVICE
  cannot both be set.

-- Configuring incomplete, errors occurred!
```

Which would indicate this section is redundant.

Reply:
Removed this part and left it to PIC to determine it.